File: QTD.SPA/src/app/_DtoModels/Version_Task_SaftyHazard_Link/Version_Task_SaftyHazard_Link.ts
import { Entity } from '../Entity';
import { Version_SaftyHazard } from '../Version_SaftyHazard/Version_SaftyHazard';
import { Version_Task } from '../Version_Task/Version_Task';

export class Version_Task_SaftyHazard_Link extends Entity {
  version_TaskId!: any;
  version_SaftyHazardId!: any;
  version_Task!: Version_Task;
  version_SaftyHazard!: Version_SaftyHazard;
}

File: QTD2.Domain/Entities/Core/Version_Task_SaftyHazard_Link.cs
namespace QTD2.Domain.Entities.Core
{
    public class Version_Task_SaftyHazard_Link : Common.Entity
    {
        public int Version_TaskId { get; set; }

        public int Version_SaftyHazardId { get; set; }

        public string VersionNumber { get; set; }

        public virtual Version_Task Version_Task { get; set; }

        public virtual Version_SaftyHazard Version_SaftyHazard { get; set; }

        public Version_Task_SaftyHazard_Link()
        {
        }

        public Version_Task_SaftyHazard_Link(int version_TaskId, int version_SaftyHazardId, string versionNumber)
        {
            Version_TaskId = version_TaskId;
            Version_SaftyHazardId = version_SaftyHazardId;
            VersionNumber = versionNumber;
        }
    }
}
